Fix issues with vet on 1.12

Running the `vet` that will come with 1.12 shows the following issues:

```
github.com/dgraph-io/badger/db.go:332:37: tautological condition: nil == nil
github.com/dgraph-io/badger/db.go:606:10: impossible condition: nil != nil
```
The first one is:
```
        // Now close the value log.
        if vlogErr := db.vlog.Close(); err == nil {
                err = errors.Wrap(vlogErr, "DB.Close")
        }
```
Where the `err` that's being checked for nil should probably be `vlogErr`.

The second one is:
```
                for err := db.ensureRoomForWrite(); err == errNoRoom; err = db.ensureRoomForWrite() {
						// trimmed
						...
                }
                if err != nil {
                        done(err)
                        return errors.Wrap(err, "writeRequests")
                }
```

Again the `err` that's being checked for nil is not the `err` that's declared with `:=` in the for loop as that variable is scoped to just the loop.

--- a/db.go
+++ b/db.go
@@ -333,7 +333,7 @@ func (db *DB) Close() (err error) {
 	db.closers.writes.SignalAndWait()
 
 	// Now close the value log.
-	if vlogErr := db.vlog.Close(); vlogErr == nil {
+	if vlogErr := db.vlog.Close(); vlogErr != nil {
 		err = errors.Wrap(vlogErr, "DB.Close")
 	}
 

--- a/db.go
+++ b/db.go
@@ -333,7 +333,7 @@ func (db *DB) Close() (err error) {
 	db.closers.writes.SignalAndWait()
 
 	// Now close the value log.
-	if vlogErr := db.vlog.Close(); err == nil {
+	if vlogErr := db.vlog.Close(); vlogErr == nil {
 		err = errors.Wrap(vlogErr, "DB.Close")
 	}
 
@@ -602,7 +602,7 @@ func (db *DB) writeRequests(reqs []*request) error {
 		}
 		count += len(b.Entries)
 		var i uint64
-		for err := db.ensureRoomForWrite(); err == errNoRoom; err = db.ensureRoomForWrite() {
+		for err = db.ensureRoomForWrite(); err == errNoRoom; err = db.ensureRoomForWrite() {
 			i++
 			if i%100 == 0 {
 				db.elog.Printf("Making room for writes")

